src/app-manager.ts

```diff
@@ -49,31 +82,97 @@ export class AppManager {
     })
   }
 
-  private broadcast(type: string, key: string, value: any) {
-    for (const origin in this.events) {
-      if (this.events[origin][type]) {
-        const destination = this.events[origin][type]
-        if (destination[key]) destination[key](value)
-      }
-    }
+  /** Activate Plugin */
+  private activatePlugin(json: PluginProfile, api: Plugin) {
+    api.request = ({ type, key, value }) => this[type][key](value)
+
+    const notifications = json.notifications || []
+    notifications.forEach(({ type, key }) => {
+      if (!this.events[api.type]) this.events[api.type] = {}
+      if (!this.events[api.type][type]) this.events[api.type][type] = {}
+      this.events[api.type][type][key] = api.notifs[type][key]
+    })
   }
 
-  /** Activate a plugin */
+  /** Activate a plugin or module */
   public activate(type: string) {
     if (!this[type]) throw new Error(`Plugin ${type} is not registered yet`)
-    this[type].activate()
-    // TODO : this.addApi()
+    if (this.plugins[type]) {
+      const { json, api } = this.plugins[type]
+      this.activateApi(json, api)
+      this.activatePlugin(json, api)

```

Comment:
that one seems to register the notifications the plugin are listening to . Am I right?
In that case that should also potentially apply to modules.
if this is not the case right now, Perhaps better to rename to `registerListenOnNotification` or something like that.

Reply:
In the current design, module api can inject other module apis (which enables types on autocompletion and better customization). That's why I didn't put the notification for module api activation. The `activatePlugin` also activates the `request` method of the plugin api. I can call it `activateRequestAndNotification` but it doesn't say that it restricted to plugin api ...
What do you think?

Reply:
ok I see, oh `activateRequestAndNotification` looks good to me.
so as i understand, if we want a module to make request to iframe plugin or a module to listen on iframe plugin event, we'll need to call `activatePlugin`, right?

Reply:
I wouldn't recommend a module to make a request to a plugin or listen to plugin events as they are not active but default, and are not maintained by the core team. It should be up to the plugin to send a request to the module. Though two plugins can communicate with each other through events.